onnxruntime/core/optimizer/transpose_optimization/onnx_transpose_optimization.cc

```diff
@@ -2586,6 +2965,56 @@ OptimizeResult OptimizeImpl(OptimizerCtx& ctx) {
     }
   }
 
+  // Run reverse 'fix up' pass for QDQ node units.
```

Comment:
nit: could we do the two fixups as lambdas or functions so there's only one loop of the graph nodes with the CanModifyNode check?  

optional: seems like the amount of code in OptimizeImpl has gotten a little out of control. could we maybe break out all the fixups into separate functions (remove empty DQ -> Q, fix transpose after DQ, Q -> Transpose graph output, Transpose inside QDQ node group), and maybe call them all from a single loop in say a 'fix qdq node units' function that is called from OptimizeImpl? can maybe look at doing that in a separate PR. would make it a bit less overwhelming to refactor it.

Reply:
I moved all QDQ fixups into separate functions. As you suggested, the `OptimizeImpl()` function now calls a single `FixQDQNodeUnits()` function that then calls other functions for each of the potential fixups.